formicarium/test/UnitTest/diff_robot_unit_test.py

```diff
@@ -0,0 +1,144 @@
+
+    def test_create_robot_posePublisher_none_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = None
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, 1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_lidar_none_throw(self):
+        lidar = None
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, 1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_wheelRadius_negative_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(-1, 1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_wheelBase_negative_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, -1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_startX_negative_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, 1, -1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_startY_negative_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, 1, 1, -1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_wheelRadius_zero_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(0, 1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_create_robot_wheelBase_zero_throw(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        with self.assertRaises(ValueError):
+            robot = DiffRobot.DiffRobot(1, 0, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
+
+    def test_GetOdometry_call_odomPublisher_publish(self):
+        lidar = Lidar_Stub()
+        odomPublisher = Publisher_Stub()
+        posePublisher = Publisher_Stub()
+
+        robot = DiffRobot.DiffRobot(1, 1, 1, 1, odomPublisher, posePublisher, lidar, 'formicarium/ant.png')
```

Comment:
ant.png?

Reply:
ant.png was an image as an alternative to the robot.png to fit the ant hill theme. It can be deleted since we have a new image now